src/NHibernate/Type/NullableType.cs

```diff
@@ -291,9 +291,10 @@ public virtual SqlType SqlType
 		/// column.  All of their implementation should be in <see cref="NullableType.SqlType" />.
 		/// </para>
 		/// </remarks>
-		public override sealed SqlType[] SqlTypes(IMapping mapping)
+		public sealed override SqlType[] SqlTypes(IMapping mapping)
 		{
-			return new SqlType[] {SqlType};
+			var types = new[] { SqlType };
+			return mapping?.Dialect.RefineSqlTypes(this, types) ?? types;
```

Comment:
That the main change for supporting the feature: this allow switching the NHibernate type default corresponding sql type for another.

Hibernate does not do that. They do not need. Java does not know DbType.DateTime and DbType.DateTime2, it only knows `Types.TIMESTAMP`. Hibernate just need to tell this one should be mapped to sql type `datetime2` in SQL 2008 [dialect](https://github.com/hibernate/hibernate-orm/blob/master/hibernate-core/src/main/java/org/hibernate/dialect/SQLServer2008Dialect.java#L29).

Reply:
They, actually, do something similar: https://github.com/hibernate/hibernate-orm/blob/9f2edb654dd55cee67d1b0445bdc0f6ae716f5e9/hibernate-core/src/main/java/org/hibernate/dialect/PostgreSQL81Dialect.java#L179

Reply:
They call it on nullSafeGet/nullSafeSet/extract(no equivalent on our side)

Reply:
Ok. It does not seems near to be ported currently, since it depends on their type descriptor which is not ported currently, and uses instance overrides for `doBind` which relies on members of the calling class, pattern not doable that way in .Net. (We could likely use method delegates instead.) And I have not checked if their override is taken into account also for hbm2ddl, as mine.